cds/modules/opencast/tasks.py

```diff
@@ -195,54 +214,49 @@ def check_transcoding_status():
             continue
 
-        for started_flow_task in started_flow_tasks:
-            for subformat in opencast_event["subformats"]:
-                is_same_preset_quality = (
-                    started_flow_task.payload["opencast_publication_tag"]
-                    in subformat["tags"]
-                )
-                if not is_same_preset_quality:
-                    continue
-
-                on_transcoding_completed.s(
-                    flow_task_id=str(started_flow_task.id),
-                    opencast_subformat=subformat,
-                    opencast_event_id=event_id
-                ).apply_async(
-                    link_error=on_celery_task_failed.s(
-                        data=dict(flow_task_id=str(started_flow_task.id))
-                    ),
-                )
-                number_of_transcoding_tasks_started += 1
-                # transcoding subformat completed, `break` to go to the next
-                break
-            else:
-                # no `break`, the transcoded subformat might not yet
-                # ready, or if workflow failed, this subformat failed
-                if opencast_event["status"] == "FAILED":
-                    msg = (
-                        "Opencast event 'processing_state' field has "
-                        "value 'FAILED' for event id: {0}.".format(event_id)
-                    )
-                    _set_flow_tasks_to_failed(
-                        [(str(started_flow_task.id), msg)]
-                    )
-    if number_of_transcoding_tasks_started:
-        current_app.logger.debug(
-            "Started {0} on_transcoding_completed tasks".format(
-                number_of_transcoding_tasks_started
+        is_failed = opencast_event["processing_state"] == "FAILED"
+        if is_failed:
+            msg = (
+                "Opencast event 'processing_state' field has "
+                "value 'FAILED' for event id: {0}.".format(event_id)
             )
+            for started_flow_task in started_flow_tasks:
+                _set_flow_tasks_to_failed(
+                    [(str(started_flow_task.id), msg)]
+                )
+            continue
+
+        processed = get_opencast_processed_subformats(
+            started_flow_tasks,
+            opencast_event["subformats"]
         )
-    else:
-        current_app.logger.debug(
-            "No on_transcoding_completed tasks started"
+        if not processed:
+            # nothing processed yet
+            continue
+
+        # We run only one subformat download at the same time.
+        # The `on_transcoding_completed` is locking on the `opencast event id`.
+
+        # This is to avoid that concurrent celery tasks will update the same deposit
+        # at the same time, creating revision id conflicts when publishing.
+
+        # This means that each subformat will be published every run of this task.
+
+        started_flow_task, opencast_subformat = processed[0]
```

Comment:
Question: Why does `processed` is a list? I would expect that it to only match with only subformat.

Reply:
The func is generic, it returns a list of matches:
```python
[
  (task_360, subformat_360),
  (task_720, subformat_720),
  ...
]
```

Reply:
Ah right, I understood it now, we pass all possible subformats and the tasks that are yet to be processed :+1: